#!/bin/sh -e
##
## 94_debian_alsaconf_modprobe_driver.dpatch by Thomas Hood
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: modprobe sound card module before finishing

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad --exclude=CVS --exclude=.svn ./alsaconf/alsaconf.in /tmp/dpep-work.tYPPRe/alsa-utils-1.0.10rc1/alsaconf/alsaconf.in
--- ./alsaconf/alsaconf.in	2005-08-24 17:00:59.000000000 +0200
+++ /tmp/dpep-work.tYPPRe/alsa-utils-1.0.10rc1/alsaconf/alsaconf.in	2005-08-24 17:02:36.000000000 +0200
@@ -465,7 +465,7 @@
 
           will prepare the card for playing now.
 
-     Now I'll run alsasound init script, then I'll use
+     Now I will load the ALSA sound driver and use
      amixer to raise the default volumes.
      You can change the volume later via a mixer
      program such as alsamixer or gamix.
@@ -890,7 +890,12 @@
       update-modules
     fi
     echo Loading driver...
+if false; then
     $rcalsasound restart
+else
+    /sbin/modprobe $CARD_DRIVER
+    sleep 4   # Wait for automatic "alsactl restore" to finish
+fi
     echo Setting default volumes...
     if [ -x $bindir/set_default_volume ]; then
 	$bindir/set_default_volume -f
